### ISSUE TITLE
Worker timeout with externalRedis

### ISSUE THREAD

**Issue by alexmbird** [getredash/contrib-helm-chart#58]
Possibly related to #34 - I've deployed Redash from the chart with settings for external PostgreSQL & Redis.

```
$ kubectl -n redash logs -f redash-7bb958dc7f-v8tfl
Using external postgresql database
Using external redis database
[2020-11-18 13:43:52 +0000] [6] [INFO] Starting gunicorn 19.7.1
[2020-11-18 13:43:52 +0000] [6] [INFO] Listening at: http://0.0.0.0:5000 (6)
[2020-11-18 13:43:52 +0000] [6] [INFO] Using worker: sync
[2020-11-18 13:43:52 +0000] [10] [INFO] Booting worker with pid: 10
[2020-11-18 13:43:52 +0000] [12] [INFO] Booting worker with pid: 12
...
[2020-11-18 13:44:32 +0000] [6] [CRITICAL] WORKER TIMEOUT (pid:10)
[2020-11-18 13:44:32 +0000] [10] [INFO] Worker exiting (pid: 10)
[2020-11-18 13:44:32 +0000] [32] [INFO] Booting worker with pid: 32
[2020-11-18 13:44:42 +0000] [6] [CRITICAL] WORKER TIMEOUT (pid:12)
[2020-11-18 13:44:42 +0000] [12] [INFO] Worker exiting (pid: 12)
[2020-11-18 13:44:42 +0000] [43] [INFO] Booting worker with pid: 43
```

As a result the pod never passes its readiness check and my Helm command eventually fails.

I've tried:

- Reducing `webWorkers` to `2` for lower resource utilisation
- Giving it a crazy-high amount of RAM as suggested in #34 (I went as high as 14Gi) 

Neither have fixed the problem.

**Comment by alexmbird** [comment]
I'm starting to think this might be because I'm using an external redis.  I tried letting the chart create its own inside Kubernetes and redash starts happily.  The problem may be that I'm using an auth token to protect the redis instance - I'm assuming the correct format for `externalRedis` is `redis://password@hostname:port/dbnum` but could be mistaken.

I've tried supplying my own env vars for `REDASH_REDIS_HOSTNAME` / `REDASH_REDIS_PORT` / `REDASH_REDIS_DB` / `REDASH_REDIS_PASSWORD` but `REDASH_REDIS_URL` is still set on the deployment, probably becuase it's switched on by `redis.enabled`.

**Comment by grugnog** [comment]
@alexmbird I am using externalRedis in production - the format you describe is correct - you should not need to provide any environment variables directly. Perhaps it could be a connectivity issue to the external Redis - could be worth trying to connect with a redis CLI?

**Comment by alexmbird** [comment]
Hi @grugnog.  I've done some further testing by spinning up an ElastiCache instance (with Redis 6.x selected in the AWS Console) and connecting to it with `redis-cli` running in a throwaway Ubuntu pod.

- Without auth configured I'm able to connect fine on port 6379 and query Redis
- With it (and transit encryption) attempts to connect fail.  Port 6379 is open but redis-cli just hangs, the same behaviour I saw with the Redash worker.

I tried downgrading the Redis instance to 5.0.6 but that shows the same behaviour.

I think the problem is to do with the transit encryption.  AWS insists on that if you want to use auth, but I've come across a knowledgebase [article](https://aws.amazon.com/premiumsupport/knowledge-center/elasticache-connect-redis-node/) that says clients don't support it.

I've fallen back to letting the Helm chart install its own Redis and all works fine now.
